test/inductor/test_torchinductor.py

```diff
@@ -10464,6 +10464,52 @@ def f(x):
             self.assertEqual(out_ref.stride(), out_test.stride())
             self.assertEqual(x_ref, x_test)
 
+    @requires_gpu()
+    @skip_if_not_triton
+    @unittest.skipIf(
+        not IS_BIG_GPU, "Skipping triton backend only since not big GPU (not enough SM)"
+    )
+    def test_inductor_multiple_specializations(self):
+        from triton.testing import do_bench
+
+        @torch.compile(
+            options={
+                "max_autotune": True,
+                "max_autotune_gemm_backends": "TRITON",
+            },
+            dynamic=False,
+        )
+        def inductor_matmul(a, b):
+            torch._check(a.shape[0] == b.shape[1])
+            return (m, torch.mm(a, b))
+
+        m = 16
+        k = 1280
+        dynamic_a = torch.randn(m, k, device="cuda", dtype=torch.bfloat16)
+        dynamic_specialized_a = torch.randn(m, k, device="cuda", dtype=torch.bfloat16)
+        b = torch.randn(k, m, device="cuda", dtype=torch.bfloat16)
+        torch._dynamo.decorators.mark_dynamic(
+            dynamic_a,
+            0,
+        )
+        torch._dynamo.decorators.mark_dynamic(
```

Comment:
what if we have multiple dynamic shapes but we can guarantee they are the same?

like https://github.com/vllm-project/vllm/blob/dc47ba32f81bb479eaee43c5b6d42e72d1cb3be4/vllm/compilation/decorators.py#L188 , we mark many tensors, this will produce multiple many symbolic ints, but they are guaranteed to be the same.

Reply:
You can use torch._check like I did above to add runtime asserts and tell our symbolic reasoning system about this fact. When one of the symbols gets specialized, they all will.